src/main.jl

```diff
@@ -391,16 +394,23 @@ function resolve(goals::Vector{<:Term}, clauses::ClauseTable{T}; options...)  wh
         # Iterate across clause set with matching heads
         matched_clauses = retrieve_clauses(clauses, term, funcs)
         matched = false
+        local_queue = MutableLinkedList{GoalTree}() # maintain a local queue for the depth first search; the new goals should be added to the global queue in this order to the front
```

Comment:
I worry about creating a local queue leading to performance slow downs because we're now allocating a queue at every iteration of the search -- is there a way to avoid using a local queue?